diff --git a/_envoy_genrule_cmd.genrule_cmd b/_envoy_genrule_cmd.genrule_cmd
index 5a37d85b0..03a0ed53d 100644
--- a/_envoy_genrule_cmd.genrule_cmd
+++ b/_envoy_genrule_cmd.genrule_cmd
@@ -2,11 +2,14 @@
 
 set -e
 
-# This works only on Linux-x86_64 and macOS-x86_64.
-if [[ ( `uname` != "Linux" && `uname` != "Darwin" ) || `uname -m` != "x86_64" ]]; then
-  echo "ERROR: wee8 is currently supported only on Linux-x86_64 and macOS-x86_64."
+# This works only on Linux-{x86_64,ppc64le,s390x,aarch64} and macOS-x86_64.
+case "$$(uname -s)-$$(uname -m)" in
+Linux-x86_64|Linux-ppc64le|Linux-s390x|Linux-aarch64|Darwin-x86_64)
+  ;;
+*)
+  echo "ERROR: wee8 is currently supported only on Linux-{x86_64,ppc64le,s390x,aarch64} and macOS-x86_64." >&2
   exit 1
-fi
+esac
 
 # Bazel magic.
 ROOT=$$(dirname $(rootpath wee8/BUILD.gn))/..
