diff -urN a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/BUILD.bazel	2019-10-05 15:46:11.320601008 -0400
@@ -0,0 +1,15 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "helper.go",
+        "validator.pb.go",
+    ],
+    importpath = "github.com/mwitkow/go-proto-validators",
+    visibility = ["//visibility:public"],
+    deps = [
+        "@com_github_gogo_protobuf//proto:go_default_library",
+        "@com_github_gogo_protobuf//protoc-gen-gogo/descriptor:go_default_library",
+    ],
+)
diff -urN a/examples/BUILD.bazel b/examples/BUILD.bazel
--- a/examples/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/examples/BUILD.bazel	2019-10-05 15:46:11.320601008 -0400
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "enum.pb.go",
+        "enum.validator.pb.go",
+        "nested.pb.go",
+        "nested.validator.pb.go",
+        "uuid.pb.go",
+        "uuid.validator.pb.go",
+    ],
+    importpath = "github.com/mwitkow/go-proto-validators/examples",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_golang_protobuf//proto:go_default_library",
+    ],
+)
diff -urN a/plugin/BUILD.bazel b/plugin/BUILD.bazel
--- a/plugin/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/plugin/BUILD.bazel	2019-10-05 15:46:11.320601008 -0400
@@ -0,0 +1,16 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["plugin.go"],
+    importpath = "github.com/mwitkow/go-proto-validators/plugin",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//:go_default_library",
+        "@com_github_gogo_protobuf//gogoproto:go_default_library",
+        "@com_github_gogo_protobuf//proto:go_default_library",
+        "@com_github_gogo_protobuf//protoc-gen-gogo/descriptor:go_default_library",
+        "@com_github_gogo_protobuf//protoc-gen-gogo/generator:go_default_library",
+        "@com_github_gogo_protobuf//vanity:go_default_library",
+    ],
+)
diff -urN a/protoc-gen-govalidators/BUILD.bazel b/protoc-gen-govalidators/BUILD.bazel
--- a/protoc-gen-govalidators/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/protoc-gen-govalidators/BUILD.bazel	2019-10-05 15:46:11.320601008 -0400
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["main.go"],
+    importpath = "github.com/mwitkow/go-proto-validators/protoc-gen-govalidators",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//plugin:go_default_library",
+        "@com_github_gogo_protobuf//proto:go_default_library",
+        "@com_github_gogo_protobuf//protoc-gen-gogo/generator:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "protoc-gen-govalidators",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/test/gogo/BUILD.bazel b/test/gogo/BUILD.bazel
--- a/test/gogo/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/test/gogo/BUILD.bazel	2019-10-05 15:46:11.320601008 -0400
@@ -0,0 +1,7 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    srcs = ["validator_test.go"],
+    deps = ["@com_github_stretchr_testify//assert:go_default_library"],
+)
diff -urN a/test/golang/BUILD.bazel b/test/golang/BUILD.bazel
--- a/test/golang/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/test/golang/BUILD.bazel	2019-10-05 15:46:11.320601008 -0400
@@ -0,0 +1,7 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_test")
+
+go_test(
+    name = "go_default_test",
+    srcs = ["validator_test.go"],
+    deps = ["@com_github_stretchr_testify//assert:go_default_library"],
+)
