diff -urN b/descriptor/BUILD.bazel c/descriptor/BUILD.bazel
--- b/descriptor/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/descriptor/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -14,6 +14,21 @@
     ],
 )
 
+go_library(
+    name = "go_default_library_gen",
+    srcs = ["descriptor.go"],
+    importpath = "github.com/golang/protobuf/descriptor",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//proto:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:descriptor_go_proto",
+        "@org_golang_google_protobuf//reflect/protodesc:go_default_library",
+        "@org_golang_google_protobuf//reflect/protoreflect:go_default_library",
+        "@org_golang_google_protobuf//runtime/protoimpl:go_default_library",
+    ],
+)
+
+# keep
 go_test(
     name = "go_default_test",
     srcs = ["descriptor_test.go"],
diff -urN b/jsonpb/BUILD.bazel c/jsonpb/BUILD.bazel
--- b/jsonpb/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/jsonpb/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -19,6 +19,12 @@
     ],
 )
 
+alias(
+    name = "go_default_library_gen",
+    actual = ":go_default_library",
+    visibility = ["//visibility:public"],
+)
+
 go_test(
     name = "go_default_test",
     srcs = ["json_test.go"],
diff -urN b/protoc-gen-go/descriptor/BUILD.bazel c/protoc-gen-go/descriptor/BUILD.bazel
--- b/protoc-gen-go/descriptor/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/protoc-gen-go/descriptor/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["descriptor.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["descriptor.pb.go"],
diff -urN b/protoc-gen-go/generator/BUILD.bazel c/protoc-gen-go/generator/BUILD.bazel
--- b/protoc-gen-go/generator/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/protoc-gen-go/generator/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -12,3 +12,16 @@
         "//protoc-gen-go/plugin:go_default_library",
     ],
 )
+
+go_library(
+    name = "go_default_library_gen",
+    srcs = ["generator.go"],
+    importpath = "github.com/golang/protobuf/protoc-gen-go/generator",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//proto:go_default_library",
+        "//protoc-gen-go/generator/internal/remap:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:compiler_plugin_go_proto",
+        "@io_bazel_rules_go//proto/wkt:descriptor_go_proto",
+    ],
+)
diff -urN b/protoc-gen-go/plugin/BUILD.bazel c/protoc-gen-go/plugin/BUILD.bazel
--- b/protoc-gen-go/plugin/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/protoc-gen-go/plugin/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["plugin.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["plugin.pb.go"],
diff -urN b/ptypes/BUILD.bazel c/ptypes/BUILD.bazel
--- b/ptypes/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -20,6 +20,26 @@
     ],
 )
 
+go_library(
+    name = "go_default_library_gen",
+    srcs = [
+        "any.go",
+        "doc.go",
+        "duration.go",
+        "timestamp.go",
+    ],
+    importpath = "github.com/golang/protobuf/ptypes",
+    visibility = ["//visibility:public"],
+    deps = [
+        "//proto:go_default_library",
+        "@io_bazel_rules_go//proto/wkt:any_go_proto",
+        "@io_bazel_rules_go//proto/wkt:duration_go_proto",
+        "@io_bazel_rules_go//proto/wkt:timestamp_go_proto",
+        "@org_golang_google_protobuf//reflect/protoreflect:go_default_library",
+        "@org_golang_google_protobuf//reflect/protoregistry:go_default_library",
+    ],
+)
+
 go_test(
     name = "go_default_test",
     srcs = [
diff -urN b/ptypes/any/BUILD.bazel c/ptypes/any/BUILD.bazel
--- b/ptypes/any/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/any/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["any.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["any.pb.go"],
diff -urN b/ptypes/duration/BUILD.bazel c/ptypes/duration/BUILD.bazel
--- b/ptypes/duration/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/duration/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["duration.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["duration.pb.go"],
diff -urN b/ptypes/empty/BUILD.bazel c/ptypes/empty/BUILD.bazel
--- b/ptypes/empty/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/empty/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["empty.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["empty.pb.go"],
diff -urN b/ptypes/struct/BUILD.bazel c/ptypes/struct/BUILD.bazel
--- b/ptypes/struct/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/struct/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["struct.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["struct.pb.go"],
diff -urN b/ptypes/timestamp/BUILD.bazel c/ptypes/timestamp/BUILD.bazel
--- b/ptypes/timestamp/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/timestamp/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["timestamp.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["timestamp.pb.go"],
diff -urN b/ptypes/wrappers/BUILD.bazel c/ptypes/wrappers/BUILD.bazel
--- b/ptypes/wrappers/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
+++ c/ptypes/wrappers/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -1,5 +1,11 @@
 load("@io_bazel_rules_go//go:def.bzl", "go_library")
 
+filegroup(
+    name = "go_default_library_protos",
+    srcs = ["wrappers.proto"],
+    visibility = ["//visibility:public"],
+)
+
 go_library(
     name = "go_default_library",
     srcs = ["wrappers.pb.go"],
