diff -urN a/BUILD.bazel b/BUILD.bazel
--- a/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,32 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "asset.go",
+        "bits.go",
+        "bytewriter.go",
+        "config.go",
+        "convert.go",
+        "debug.go",
+        "doc.go",
+        "release.go",
+        "restore.go",
+        "safefile.go",
+        "stringwriter.go",
+        "toc.go",
+    ],
+    importpath = "github.com/kevinburke/go-bindata",
+    visibility = ["//visibility:public"],
+)
+
+go_test(
+    name = "go_default_test",
+    srcs = [
+        "benchmark_test.go",
+        "convert_test.go",
+        "release_test.go",
+        "safefile_test.go",
+    ],
+    embed = [":go_default_library"],
+)
diff -urN a/go-bindata/BUILD.bazel b/go-bindata/BUILD.bazel
--- a/go-bindata/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/go-bindata/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,19 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "AppendSliceValue.go",
+        "main.go",
+        "version.go",
+    ],
+    importpath = "github.com/kevinburke/go-bindata/go-bindata",
+    visibility = ["//visibility:private"],
+    deps = ["//:go_default_library"],
+)
+
+go_binary(
+    name = "go-bindata",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
diff -urN a/testdata/assets/BUILD.bazel b/testdata/assets/BUILD.bazel
--- a/testdata/assets/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/testdata/assets/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["bindata.go"],
+    importpath = "github.com/kevinburke/go-bindata/testdata/assets",
+    visibility = ["//visibility:public"],
+)
diff -urN a/testdata/out/BUILD.bazel b/testdata/out/BUILD.bazel
--- a/testdata/out/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ b/testdata/out/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,20 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = [
+        "compress-memcopy.go",
+        "compress-nomemcopy.go",
+        "debug.go",
+        "nocompress-memcopy.go",
+        "nocompress-nomemcopy.go",
+    ],
+    importpath = "github.com/kevinburke/go-bindata/testdata/out",
+    visibility = ["//visibility:private"],
+)
+
+go_binary(
+    name = "out",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
