diff -urN b/BUILD.bazel c/BUILD.bazel
--- b/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ c/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1,4 @@
+# gazelle:proto package
+# gazelle:proto_group go_package
+# gazelle:exclude gapic
+# gazelle:exclude third_party
diff -urN b/google/BUILD.bazel c/google/BUILD.bazel
--- b/google/BUILD.bazel	1969-12-31 19:00:00.000000000 -0500
+++ c/google/BUILD.bazel	2000-01-01 00:00:00.000000000 -0000
@@ -0,0 +1 @@
+# gazelle:prefix google.golang.org/genproto/googleapis
